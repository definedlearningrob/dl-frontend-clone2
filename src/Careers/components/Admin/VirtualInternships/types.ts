import { TOpportunity as TBaseOpportunity } from '@dc/resources/types';

export const enum VirtualInternshipStatuses {
  DRAFT = 'DRAFT',
  PUBLISHED = 'PUBLISHED',
}

export type VirtualInternshipLessonAttributes = {
  lessonId: string;
  step: number;
};

export type FormValues = {
  badges: {
    id: string;
    name: string;
    imageUrl: string;
  }[];
  description: string;
  name: string;
  pathways: {
    label: string;
    value: string;
  }[];
  availableSpots: number;
  creditsOutcomes: string;
  image: Record<string, unknown>;
  tags: {
    label: string;
    value: string;
  }[];
  requiredExperiences: number;
  status: { value: VirtualInternshipStatuses; label: string };
  calendarLessons: Lesson[];
  experienceOpportunityLessons: Lesson[];
  postExperienceLessons: Lesson[];
  readinessSkillsLessons: Lesson[];
};

export type TPathway = Pick<TBaseOpportunity['pathways'][number], 'id' | 'name'>;

export type TOpportunity = Pick<
  TBaseOpportunity,
  | 'id'
  | 'name'
  | 'availableSpots'
  | 'creditsOutcomes'
  | 'description'
  | 'imageUrl'
  | 'opportunityType'
  | 'tags'
  | 'salaryInformation'
> & {
  pathways: TPathway[];
};

export type Lesson = {
  id: string;
  step: number;
  name: string;
  imageUrl: string;
  thumbnailUrl: string;
  type: string;
};

export type TVirtualInternship = {
  archivedAt: string | null;
  badges: {
    id: string;
    name: string;
    imageUrl: string;
  }[];
  id: string;
  opportunity: TOpportunity;
  requiredExperiences: number;
  status: VirtualInternshipStatuses;
  calendarLessons: Lesson[];
  experienceOpportunityLessons: Lesson[];
  postExperienceLessons: Lesson[];
  readinessSkillsLessons: Lesson[];
};
