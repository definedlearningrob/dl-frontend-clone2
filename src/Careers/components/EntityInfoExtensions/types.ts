export type TAdminEntityInfoExtensionField = {
  id: string;
  name: string;
};
