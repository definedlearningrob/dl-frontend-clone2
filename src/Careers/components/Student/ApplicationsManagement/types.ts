export type Counselor = {
  email: string;
  firstName: string;
  fullName: string;
  lastName: string;
  name: string;
  username: string;
  uuid: string;
};

export type AvailableTeacher = {
  email: string;
  fullName: string;
  name: string;
  username: string;
  uuid: string;
};

export type CounselorSelectOption = { label: string; value: string };
