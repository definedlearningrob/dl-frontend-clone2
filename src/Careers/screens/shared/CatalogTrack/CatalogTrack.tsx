import { useCatalogTrackQuery } from '@graphql/dc/shared/hooks';
import { useEffect } from 'react';
import { useParams } from 'react-router';

import { useNavigation } from '@shared/components/Sidebar/useNavigation';
import { Track } from '@shared/components/Track/Track';

export const CatalogTrack = () => {
  const { setBackNavButton } = useNavigation();
  const { id } = useParams<{ id: string }>();
  const { data, loading } = useCatalogTrackQuery({ variables: { id } });

  useEffect(() => {
    setBackNavButton(true);

    return () => {
      setBackNavButton(false);
    };
  }, []);

  return <Track isLoading={loading} track={data?.careersCatalog?.track} />;
};
