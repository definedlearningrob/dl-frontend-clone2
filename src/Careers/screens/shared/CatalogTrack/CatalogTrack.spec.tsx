import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { CatalogTrackQuery } from '@graphql/dc/shared/operations';
import { CatalogTrackDocument } from '@graphql/dc/shared/hooks';
import { UnitResourceTypes } from '@graphql/dc/shared/types';
import { screen, within } from '@testing-library/dom';
import userEvent from '@testing-library/user-event';
import { Route } from 'react-router';

import { NavigationContextProvider } from '@shared/components/Sidebar/useNavigation';
import { renderWithRouter } from '@shared/utils/test';

import { CatalogTrack } from './CatalogTrack';

const trackMock: MockedResponse<CatalogTrackQuery> = {
  request: {
    query: CatalogTrackDocument,
    variables: { id: '1' },
  },
  result: {
    data: {
      careersCatalog: {
        id: '11',
        track: {
          id: '1',
          name: 'Track',
          description: 'Track description',
          imageUrl: 'https://via.placeholder.com/150',
          thumbnailUrl: 'https://via.placeholder.com/150',
          grades: ['K', '1'],
          resourcesCount: 5,
          units: [
            {
              id: '111',
              name: 'Unit 1',
              description: 'Unit 1 description',
              imageUrl: 'https://via.placeholder.com/150',
              thumbnailUrl: 'https://via.placeholder.com/150',
              resources: [
                {
                  resourceId: '1111',
                  resourceType: UnitResourceTypes.COURSE,
                  description: 'Course 1 description',
                  imageUrl: 'https://via.placeholder.com/150',
                  thumbnailUrl: 'https://via.placeholder.com/150',
                  name: 'Course 1',
                  pathways: [{ name: 'Marketing Research' }],
                  isVirtualInternship: null,
                },
                {
                  resourceId: '1112',
                  resourceType: UnitResourceTypes.OPPORTUNITY,
                  description: 'Opportunity 1 description',
                  imageUrl: 'https://via.placeholder.com/150',
                  thumbnailUrl: 'https://via.placeholder.com/150',
                  name: 'Opportunity 1',
                  pathways: [{ name: 'Animal Systems' }, { name: 'Printing Technology' }],
                  isVirtualInternship: false,
                },
                {
                  resourceId: '1113',
                  resourceType: UnitResourceTypes.OPPORTUNITY,
                  description: 'Virtual Internship 1 description',
                  imageUrl: 'https://via.placeholder.com/150',
                  thumbnailUrl: 'https://via.placeholder.com/150',
                  name: 'Virtual Internship 1',
                  pathways: [{ name: 'Environmental Service Systems' }],
                  isVirtualInternship: true,
                },
              ],
            },
            {
              id: '112',
              name: 'Unit 2',
              description: 'Unit 2 description',
              imageUrl: 'https://via.placeholder.com/150',
              thumbnailUrl: 'https://via.placeholder.com/150',
              resources: [
                {
                  resourceId: '1114',
                  resourceType: UnitResourceTypes.COURSE,
                  description: 'Course 2 description',
                  imageUrl: 'https://via.placeholder.com/150',
                  thumbnailUrl: 'https://via.placeholder.com/150',
                  name: 'Course 2',
                  pathways: [{ name: 'Facility and Mobile Equipment Maintenance' }],
                  isVirtualInternship: null,
                },
                {
                  resourceId: '1115',
                  resourceType: UnitResourceTypes.OPPORTUNITY,
                  description: 'Opportunity 2 description',
                  imageUrl: 'https://via.placeholder.com/150',
                  thumbnailUrl: 'https://via.placeholder.com/150',
                  name: 'Opportunity 2',
                  pathways: [{ name: 'Transportation Operations' }],
                  isVirtualInternship: false,
                },
              ],
            },
          ],
        },
      },
    },
  },
};

const renderComponent = () =>
  renderWithRouter(
    <MockedProvider mocks={[trackMock]}>
      <NavigationContextProvider>
        <Route path='/catalog/unit-outline/:id'>
          <CatalogTrack />
        </Route>
      </NavigationContextProvider>
    </MockedProvider>,
    {
      route: '/catalog/unit-outline/1',
    }
  );

describe('CatalogTrack', () => {
  it('renders correctly', async () => {
    const { container } = renderComponent();

    const trackName = await screen.findByRole('heading', { name: 'Track' });
    const trackDescription = screen.getByText('Track description');

    expect(trackName).toBeInTheDocument();
    expect(trackDescription).toBeInTheDocument();

    const unitList = await screen.findByRole('list', { name: 'Unit list' });
    const unitSections = within(unitList).getAllByRole('listitem', { name: /Unit/i });

    expect(unitSections).toHaveLength(2);

    expect(unitSections[0]).toHaveTextContent('Unit 1');
    expect(unitSections[1]).toHaveTextContent('Unit 2');

    const unitResources = within(unitSections[0]).getAllByRole('listitem');
    expect(unitResources).toHaveLength(3);
    expect(unitResources[0]).toHaveTextContent('Course 1');
    expect(unitResources[1]).toHaveTextContent('Opportunity 1');
    expect(unitResources[2]).toHaveTextContent('Virtual Internship 1');

    const unitResources2 = within(unitSections[1]).getAllByRole('listitem');
    expect(unitResources2).toHaveLength(2);
    expect(unitResources2[0]).toHaveTextContent('Course 2');
    expect(unitResources2[1]).toHaveTextContent('Opportunity 2');

    expect(container).toMatchSnapshot();
  });

  it('navigates to a resource after clicking on the card', async () => {
    const { history } = renderComponent();
    history.push = jest.fn();

    const courseCard = await screen.findByRole('link', { name: 'Course 1' });
    userEvent.click(courseCard);

    expect(history.push).toHaveBeenCalledTimes(1);
    expect(history.push).toHaveBeenCalledWith('/courses/1111');

    // @ts-ignore
    history.push.mockClear();

    const opportunityCard = await screen.findByRole('link', { name: 'Opportunity 1' });
    userEvent.click(opportunityCard);

    expect(history.push).toHaveBeenCalledTimes(1);
    expect(history.push).toHaveBeenCalledWith('/opportunities/1112');

    // @ts-ignore
    history.push.mockClear();

    const virtualInternshipCard = await screen.findByRole('link', {
      name: 'Virtual Internship 1',
    });
    userEvent.click(virtualInternshipCard);

    expect(history.push).toHaveBeenCalledTimes(1);
    expect(history.push).toHaveBeenCalledWith('/opportunities/1113');
  });

  it('navigates to a resource after clicking on the navigation item', async () => {
    const { history } = renderComponent();
    history.push = jest.fn();

    const trackNavigation = await screen.findByTestId('table-of-content');
    const units = within(trackNavigation).getAllByRole('button');

    expect(units).toHaveLength(2);

    const [firstResourceList] = within(trackNavigation).getAllByRole('list');
    const resourceItems = within(firstResourceList).getAllByRole('link');
    expect(resourceItems).toHaveLength(3);
    expect(resourceItems[0]).toHaveTextContent('Course 1');
    expect(resourceItems[1]).toHaveTextContent('Opportunity 1');
    expect(resourceItems[2]).toHaveTextContent('Virtual Internship 1');

    userEvent.click(resourceItems[0]);
    expect(history.push).toHaveBeenCalledWith('/courses/1111');

    // @ts-ignore
    history.push.mockClear();

    userEvent.click(resourceItems[1]);
    expect(history.push).toHaveBeenCalledWith('/opportunities/1112');

    // @ts-ignore
    history.push.mockClear();

    userEvent.click(resourceItems[2]);
    expect(history.push).toHaveBeenCalledWith('/opportunities/1113');
  });
});
