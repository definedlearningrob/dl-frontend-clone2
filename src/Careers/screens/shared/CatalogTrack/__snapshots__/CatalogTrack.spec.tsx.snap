// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogTrack renders correctly 1`] = `
<div>
  <div
    class="p-base xxxl:p-md flex gap-base xxxl:gap-md pt-xs xxxl:pt-sm"
    id="read"
  >
    <div
      class="flex-1 flex flex-col gap-base xxxl:gap-md"
    >
      <div
        class="card flex gap-base xxxl:gap-md"
        data-testid="track-summary-card"
      >
        <img
          class="rounded-sm h-[120px] xxxl:h-[180px] w-[213px] xxxl:w-[320px] shrink-0 object-cover"
          src="https://via.placeholder.com/150"
        />
        <div
          class="flex flex-col gap-sm"
        >
          <div
            class="flex flex-col gap-xs"
          >
            <div
              class="flex gap-xs xxxl:gap-sm"
            >
              <span
                class="min-w-0"
                data-state="closed"
              >
                <div
                  class="flex gap-xxs px-xs py-xxs rounded-full text-xxs font-medium leading-lg transition-all bg-neutral-200 text-font-primary"
                >
                  <div
                    class="truncate"
                  >
                    Grades: K - 1
                  </div>
                </div>
              </span>
              <div
                class="flex gap-xxs px-xs py-xxs rounded-full text-xxs font-medium leading-lg transition-all bg-neutral-200 text-font-primary"
              >
                <div
                  class="truncate"
                >
                  Projects: 5
                </div>
              </div>
            </div>
            <h5
              class="mb-0 text-sm xxxl:text-base"
            >
              Track
            </h5>
          </div>
          <div
            class="injected-content text-xxs xxxl:text-xs leading-lg text-font-secondary line-clamp-3"
          >
            Track description
          </div>
        </div>
      </div>
      <ul
        aria-label="Unit list"
        class="flex flex-col gap-sm"
      >
        <li
          aria-label="Unit 1"
        >
          <div
            class="card"
            id="unit-111"
          >
            <div
              class="flex gap-base mb-base"
            >
              <img
                class="rounded-sm h-[80px] xxxl:h-[124px] w-[142px] xxxl:w-[220px] shrink-0 object-cover"
                src="https://via.placeholder.com/150"
              />
              <div>
                <div
                  class="kicker sm default"
                >
                  Unit 1
                </div>
                <h6
                  class="text-xs xxxl:text-sm mb-xs xxxl:mb-sm"
                >
                  Unit 1
                </h6>
                <div
                  class="injected-content text-xxs xxxl:text-xs leading-lg text-font-secondary"
                >
                  Unit 1 description
                </div>
              </div>
            </div>
            <div>
              <div
                class="flex gap-sm items-center text-xs xxxl:text-sm font-bold after:block after:w-full after:h-[1px] after:bg-neutral-300"
              >
                <span
                  class="whitespace-nowrap"
                >
                  Projects 
                  <span
                    class="text-neutral-600"
                  >
                    (3)
                  </span>
                </span>
              </div>
              <ul
                class="grid grid-cols-2 xxl:grid-cols-3 gap-sm pt-sm"
              >
                <li>
                  <a
                    class="group h-full flex flex-col rounded-sm text-font-primary hover:text-font-primary bg-white border border-neutral-300 hover:border-neutral-400 hover:shadow-300 transition-shadow"
                    href="/courses/1111"
                  >
                    <div
                      class="relative before:bg-neutral-800 before:opacity-[48%] before:absolute before:inset-0 before:rounded-t-sm group-hover:before:opacity-0 before:transition-opacity"
                    >
                      <img
                        class="rounded-t-sm h-[180px] xxxl:h-[192px] w-full object-cover"
                        src="https://via.placeholder.com/150"
                      />
                      <span
                        class="absolute top-sm left-sm"
                        data-state="closed"
                      >
                        <div
                          class="flex"
                        >
                          <div
                            class="flex align-center justify-center grow-0 shrink-0 rounded-sm bg-white text-primary-500 p-xs"
                            data-testid="icon"
                          >
                            <div
                              class="h-base w-base"
                            />
                          </div>
                        </div>
                      </span>
                    </div>
                    <div
                      class="flex-1 py-xs xxxl:py-sm px-sm border-b border-neutral-300"
                    >
                      <div
                        class="mb-xs max-w-fit"
                      >
                        <span
                          data-state="closed"
                        >
                          <div
                            class="flex gap-xxs px-xs py-xxs rounded-full text-xxs font-medium leading-lg transition-all bg-neutral-200 text-font-primary"
                          >
                            <div
                              class="flex"
                            >
                              <div
                                class="flex align-center justify-center grow-0 shrink-0"
                                data-testid="icon"
                              >
                                <div
                                  class="h-sm w-sm"
                                />
                              </div>
                            </div>
                            <div
                              class="truncate"
                            >
                              Marketing Research
                            </div>
                          </div>
                        </span>
                      </div>
                      <span
                        data-state="closed"
                      >
                        <h6
                          class="text-xs xxxl:text-sm line-clamp-2"
                        >
                          Course 1
                        </h6>
                      </span>
                      <div
                        class="injected-content text-xxs xxxl:text-xs leading-lg text-font-secondary group-hover:text-font-primary transition-colors line-clamp-4 xxxl:line-clamp-6"
                      >
                        Course 1 description
                      </div>
                    </div>
                    <div
                      class="flex items-center gap-xxs group-hover:gap-xs p-sm text-primary-500 group-hover:text-primary-600 transition-all"
                    >
                      <span
                        class="text-xs xxxl:text-sm leading-lg font-medium"
                      >
                        View
                      </span>
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0"
                          data-testid="icon"
                        >
                          <div
                            class="h-sm w-sm"
                          />
                        </div>
                      </div>
                    </div>
                  </a>
                </li>
                <li>
                  <a
                    class="group h-full flex flex-col rounded-sm text-font-primary hover:text-font-primary bg-white border border-neutral-300 hover:border-neutral-400 hover:shadow-300 transition-shadow"
                    href="/opportunities/1112"
                  >
                    <div
                      class="relative before:bg-neutral-800 before:opacity-[48%] before:absolute before:inset-0 before:rounded-t-sm group-hover:before:opacity-0 before:transition-opacity"
                    >
                      <img
                        class="rounded-t-sm h-[180px] xxxl:h-[192px] w-full object-cover"
                        src="https://via.placeholder.com/150"
                      />
                      <span
                        class="absolute top-sm left-sm"
                        data-state="closed"
                      >
                        <div
                          class="flex"
                        >
                          <div
                            class="flex align-center justify-center grow-0 shrink-0 rounded-sm bg-white text-primary-500 p-xs"
                            data-testid="icon"
                          >
                            <div
                              class="h-base w-base"
                            />
                          </div>
                        </div>
                      </span>
                    </div>
                    <div
                      class="flex-1 py-xs xxxl:py-sm px-sm border-b border-neutral-300"
                    >
                      <div
                        class="mb-xs max-w-fit"
                      >
                        <span
                          data-state="closed"
                        >
                          <div
                            class="flex gap-xxs px-xs py-xxs rounded-full text-xxs font-medium leading-lg transition-all bg-neutral-200 text-font-primary"
                          >
                            <div
                              class="flex"
                            >
                              <div
                                class="flex align-center justify-center grow-0 shrink-0"
                                data-testid="icon"
                              >
                                <div
                                  class="h-sm w-sm"
                                />
                              </div>
                            </div>
                            <div
                              class="truncate"
                            >
                              Animal Systems, Printing Technology
                            </div>
                          </div>
                        </span>
                      </div>
                      <span
                        data-state="closed"
                      >
                        <h6
                          class="text-xs xxxl:text-sm line-clamp-2"
                        >
                          Opportunity 1
                        </h6>
                      </span>
                      <div
                        class="injected-content text-xxs xxxl:text-xs leading-lg text-font-secondary group-hover:text-font-primary transition-colors line-clamp-4 xxxl:line-clamp-6"
                      >
                        Opportunity 1 description
                      </div>
                    </div>
                    <div
                      class="flex items-center gap-xxs group-hover:gap-xs p-sm text-primary-500 group-hover:text-primary-600 transition-all"
                    >
                      <span
                        class="text-xs xxxl:text-sm leading-lg font-medium"
                      >
                        View
                      </span>
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0"
                          data-testid="icon"
                        >
                          <div
                            class="h-sm w-sm"
                          />
                        </div>
                      </div>
                    </div>
                  </a>
                </li>
                <li>
                  <a
                    class="group h-full flex flex-col rounded-sm text-font-primary hover:text-font-primary bg-white border border-neutral-300 hover:border-neutral-400 hover:shadow-300 transition-shadow"
                    href="/opportunities/1113"
                  >
                    <div
                      class="relative before:bg-neutral-800 before:opacity-[48%] before:absolute before:inset-0 before:rounded-t-sm group-hover:before:opacity-0 before:transition-opacity"
                    >
                      <img
                        class="rounded-t-sm h-[180px] xxxl:h-[192px] w-full object-cover"
                        src="https://via.placeholder.com/150"
                      />
                      <span
                        class="absolute top-sm left-sm"
                        data-state="closed"
                      >
                        <div
                          class="flex"
                        >
                          <div
                            class="flex align-center justify-center grow-0 shrink-0 rounded-sm bg-white text-primary-500 p-xs"
                            data-testid="icon"
                          >
                            <div
                              class="h-base w-base"
                            />
                          </div>
                        </div>
                      </span>
                    </div>
                    <div
                      class="flex-1 py-xs xxxl:py-sm px-sm border-b border-neutral-300"
                    >
                      <div
                        class="mb-xs max-w-fit"
                      >
                        <span
                          data-state="closed"
                        >
                          <div
                            class="flex gap-xxs px-xs py-xxs rounded-full text-xxs font-medium leading-lg transition-all bg-neutral-200 text-font-primary"
                          >
                            <div
                              class="flex"
                            >
                              <div
                                class="flex align-center justify-center grow-0 shrink-0"
                                data-testid="icon"
                              >
                                <div
                                  class="h-sm w-sm"
                                />
                              </div>
                            </div>
                            <div
                              class="truncate"
                            >
                              Environmental Service Systems
                            </div>
                          </div>
                        </span>
                      </div>
                      <span
                        data-state="closed"
                      >
                        <h6
                          class="text-xs xxxl:text-sm line-clamp-2"
                        >
                          Virtual Internship 1
                        </h6>
                      </span>
                      <div
                        class="injected-content text-xxs xxxl:text-xs leading-lg text-font-secondary group-hover:text-font-primary transition-colors line-clamp-4 xxxl:line-clamp-6"
                      >
                        Virtual Internship 1 description
                      </div>
                    </div>
                    <div
                      class="flex items-center gap-xxs group-hover:gap-xs p-sm text-primary-500 group-hover:text-primary-600 transition-all"
                    >
                      <span
                        class="text-xs xxxl:text-sm leading-lg font-medium"
                      >
                        View
                      </span>
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0"
                          data-testid="icon"
                        >
                          <div
                            class="h-sm w-sm"
                          />
                        </div>
                      </div>
                    </div>
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </li>
        <li
          aria-label="Unit 2"
        >
          <div
            class="card"
            id="unit-112"
          >
            <div
              class="flex gap-base mb-base"
            >
              <img
                class="rounded-sm h-[80px] xxxl:h-[124px] w-[142px] xxxl:w-[220px] shrink-0 object-cover"
                src="https://via.placeholder.com/150"
              />
              <div>
                <div
                  class="kicker sm default"
                >
                  Unit 2
                </div>
                <h6
                  class="text-xs xxxl:text-sm mb-xs xxxl:mb-sm"
                >
                  Unit 2
                </h6>
                <div
                  class="injected-content text-xxs xxxl:text-xs leading-lg text-font-secondary"
                >
                  Unit 2 description
                </div>
              </div>
            </div>
            <div>
              <div
                class="flex gap-sm items-center text-xs xxxl:text-sm font-bold after:block after:w-full after:h-[1px] after:bg-neutral-300"
              >
                <span
                  class="whitespace-nowrap"
                >
                  Projects 
                  <span
                    class="text-neutral-600"
                  >
                    (2)
                  </span>
                </span>
              </div>
              <ul
                class="grid grid-cols-2 xxl:grid-cols-3 gap-sm pt-sm"
              >
                <li>
                  <a
                    class="group h-full flex flex-col rounded-sm text-font-primary hover:text-font-primary bg-white border border-neutral-300 hover:border-neutral-400 hover:shadow-300 transition-shadow"
                    href="/courses/1114"
                  >
                    <div
                      class="relative before:bg-neutral-800 before:opacity-[48%] before:absolute before:inset-0 before:rounded-t-sm group-hover:before:opacity-0 before:transition-opacity"
                    >
                      <img
                        class="rounded-t-sm h-[180px] xxxl:h-[192px] w-full object-cover"
                        src="https://via.placeholder.com/150"
                      />
                      <span
                        class="absolute top-sm left-sm"
                        data-state="closed"
                      >
                        <div
                          class="flex"
                        >
                          <div
                            class="flex align-center justify-center grow-0 shrink-0 rounded-sm bg-white text-primary-500 p-xs"
                            data-testid="icon"
                          >
                            <div
                              class="h-base w-base"
                            />
                          </div>
                        </div>
                      </span>
                    </div>
                    <div
                      class="flex-1 py-xs xxxl:py-sm px-sm border-b border-neutral-300"
                    >
                      <div
                        class="mb-xs max-w-fit"
                      >
                        <span
                          data-state="closed"
                        >
                          <div
                            class="flex gap-xxs px-xs py-xxs rounded-full text-xxs font-medium leading-lg transition-all bg-neutral-200 text-font-primary"
                          >
                            <div
                              class="flex"
                            >
                              <div
                                class="flex align-center justify-center grow-0 shrink-0"
                                data-testid="icon"
                              >
                                <div
                                  class="h-sm w-sm"
                                />
                              </div>
                            </div>
                            <div
                              class="truncate"
                            >
                              Facility and Mobile Equipment Maintenance
                            </div>
                          </div>
                        </span>
                      </div>
                      <span
                        data-state="closed"
                      >
                        <h6
                          class="text-xs xxxl:text-sm line-clamp-2"
                        >
                          Course 2
                        </h6>
                      </span>
                      <div
                        class="injected-content text-xxs xxxl:text-xs leading-lg text-font-secondary group-hover:text-font-primary transition-colors line-clamp-4 xxxl:line-clamp-6"
                      >
                        Course 2 description
                      </div>
                    </div>
                    <div
                      class="flex items-center gap-xxs group-hover:gap-xs p-sm text-primary-500 group-hover:text-primary-600 transition-all"
                    >
                      <span
                        class="text-xs xxxl:text-sm leading-lg font-medium"
                      >
                        View
                      </span>
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0"
                          data-testid="icon"
                        >
                          <div
                            class="h-sm w-sm"
                          />
                        </div>
                      </div>
                    </div>
                  </a>
                </li>
                <li>
                  <a
                    class="group h-full flex flex-col rounded-sm text-font-primary hover:text-font-primary bg-white border border-neutral-300 hover:border-neutral-400 hover:shadow-300 transition-shadow"
                    href="/opportunities/1115"
                  >
                    <div
                      class="relative before:bg-neutral-800 before:opacity-[48%] before:absolute before:inset-0 before:rounded-t-sm group-hover:before:opacity-0 before:transition-opacity"
                    >
                      <img
                        class="rounded-t-sm h-[180px] xxxl:h-[192px] w-full object-cover"
                        src="https://via.placeholder.com/150"
                      />
                      <span
                        class="absolute top-sm left-sm"
                        data-state="closed"
                      >
                        <div
                          class="flex"
                        >
                          <div
                            class="flex align-center justify-center grow-0 shrink-0 rounded-sm bg-white text-primary-500 p-xs"
                            data-testid="icon"
                          >
                            <div
                              class="h-base w-base"
                            />
                          </div>
                        </div>
                      </span>
                    </div>
                    <div
                      class="flex-1 py-xs xxxl:py-sm px-sm border-b border-neutral-300"
                    >
                      <div
                        class="mb-xs max-w-fit"
                      >
                        <span
                          data-state="closed"
                        >
                          <div
                            class="flex gap-xxs px-xs py-xxs rounded-full text-xxs font-medium leading-lg transition-all bg-neutral-200 text-font-primary"
                          >
                            <div
                              class="flex"
                            >
                              <div
                                class="flex align-center justify-center grow-0 shrink-0"
                                data-testid="icon"
                              >
                                <div
                                  class="h-sm w-sm"
                                />
                              </div>
                            </div>
                            <div
                              class="truncate"
                            >
                              Transportation Operations
                            </div>
                          </div>
                        </span>
                      </div>
                      <span
                        data-state="closed"
                      >
                        <h6
                          class="text-xs xxxl:text-sm line-clamp-2"
                        >
                          Opportunity 2
                        </h6>
                      </span>
                      <div
                        class="injected-content text-xxs xxxl:text-xs leading-lg text-font-secondary group-hover:text-font-primary transition-colors line-clamp-4 xxxl:line-clamp-6"
                      >
                        Opportunity 2 description
                      </div>
                    </div>
                    <div
                      class="flex items-center gap-xxs group-hover:gap-xs p-sm text-primary-500 group-hover:text-primary-600 transition-all"
                    >
                      <span
                        class="text-xs xxxl:text-sm leading-lg font-medium"
                      >
                        View
                      </span>
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0"
                          data-testid="icon"
                        >
                          <div
                            class="h-sm w-sm"
                          />
                        </div>
                      </div>
                    </div>
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </li>
      </ul>
    </div>
    <div
      class="card withoutPadding basis-[318px] xxxl:basis-[512px] grow-0 shrink-0 max-h-[calc(theme(layout.containerHeight)-32px)] xxxl:max-h-[calc(theme(layout.containerHeight)-48px)] scrollbar sticky top-[56px] xxxl:top-[64px]"
      data-testid="table-of-content"
    >
      <div
        class="p-base pb-sm xxxl:p-md bg-white sticky top-0 z-lower"
      >
        <h5
          class="text-sm xxxl:text-base mb-0"
        >
          Unit Outline
        </h5>
      </div>
      <div
        class="flex flex-col gap-md xxxl:gap-lg p-base xxxl:p-md !pt-0"
      >
        <div
          class="relative flex flex-col gap-xs xxxl:gap-sm after:w-full after:h-[1px] after:bg-neutral-300 after:absolute after:-bottom-sm after:xxxl:-bottom-base last:after:content-none"
        >
          <button
            class="flex gap-xs text-left"
          >
            <div
              class="flex"
            >
              <div
                class="flex align-center justify-center grow-0 shrink-0 bg-neutral-200 rounded-sm text-primary-500 p-xs"
                data-testid="icon"
              >
                <div
                  class="h-sm w-sm"
                />
              </div>
            </div>
            <div>
              <div
                class="kicker sm dark !mb-xxs xxxl:!mb-xs"
              >
                Unit 1
              </div>
              <h6
                class="text-xs xxxl:text-sm mb-0 text-primary-500"
              >
                Unit 1
              </h6>
            </div>
          </button>
          <ul
            class="pt-xs px-base xxxl:px-md flex flex-col gap-xs xxxl:gap-x"
          >
            <li>
              <a
                class="flex items-center gap-xs"
                href="/courses/1111"
              >
                <div
                  class="flex"
                >
                  <div
                    class="flex align-center justify-center grow-0 shrink-0 rounded-xs bg-primary-200 text-primary-500 p-xxs"
                    data-testid="icon"
                  >
                    <div
                      class="h-sm w-sm"
                    />
                  </div>
                </div>
                <span
                  data-state="closed"
                >
                  <span
                    class="text-xxs xxxl:text-xs font-medium leading-lg line-clamp-2"
                  >
                    Course 1
                  </span>
                </span>
              </a>
            </li>
            <li>
              <a
                class="flex items-center gap-xs"
                href="/opportunities/1112"
              >
                <div
                  class="flex"
                >
                  <div
                    class="flex align-center justify-center grow-0 shrink-0 rounded-xs bg-primary-200 text-primary-500 p-xxs"
                    data-testid="icon"
                  >
                    <div
                      class="h-sm w-sm"
                    />
                  </div>
                </div>
                <span
                  data-state="closed"
                >
                  <span
                    class="text-xxs xxxl:text-xs font-medium leading-lg line-clamp-2"
                  >
                    Opportunity 1
                  </span>
                </span>
              </a>
            </li>
            <li>
              <a
                class="flex items-center gap-xs"
                href="/opportunities/1113"
              >
                <div
                  class="flex"
                >
                  <div
                    class="flex align-center justify-center grow-0 shrink-0 rounded-xs bg-primary-200 text-primary-500 p-xxs"
                    data-testid="icon"
                  >
                    <div
                      class="h-sm w-sm"
                    />
                  </div>
                </div>
                <span
                  data-state="closed"
                >
                  <span
                    class="text-xxs xxxl:text-xs font-medium leading-lg line-clamp-2"
                  >
                    Virtual Internship 1
                  </span>
                </span>
              </a>
            </li>
          </ul>
        </div>
        <div
          class="relative flex flex-col gap-xs xxxl:gap-sm after:w-full after:h-[1px] after:bg-neutral-300 after:absolute after:-bottom-sm after:xxxl:-bottom-base last:after:content-none"
        >
          <button
            class="flex gap-xs text-left"
          >
            <div
              class="flex"
            >
              <div
                class="flex align-center justify-center grow-0 shrink-0 bg-neutral-200 rounded-sm text-primary-500 p-xs"
                data-testid="icon"
              >
                <div
                  class="h-sm w-sm"
                />
              </div>
            </div>
            <div>
              <div
                class="kicker sm dark !mb-xxs xxxl:!mb-xs"
              >
                Unit 2
              </div>
              <h6
                class="text-xs xxxl:text-sm mb-0 text-primary-500"
              >
                Unit 2
              </h6>
            </div>
          </button>
          <ul
            class="pt-xs px-base xxxl:px-md flex flex-col gap-xs xxxl:gap-x"
          >
            <li>
              <a
                class="flex items-center gap-xs"
                href="/courses/1114"
              >
                <div
                  class="flex"
                >
                  <div
                    class="flex align-center justify-center grow-0 shrink-0 rounded-xs bg-primary-200 text-primary-500 p-xxs"
                    data-testid="icon"
                  >
                    <div
                      class="h-sm w-sm"
                    />
                  </div>
                </div>
                <span
                  data-state="closed"
                >
                  <span
                    class="text-xxs xxxl:text-xs font-medium leading-lg line-clamp-2"
                  >
                    Course 2
                  </span>
                </span>
              </a>
            </li>
            <li>
              <a
                class="flex items-center gap-xs"
                href="/opportunities/1115"
              >
                <div
                  class="flex"
                >
                  <div
                    class="flex align-center justify-center grow-0 shrink-0 rounded-xs bg-primary-200 text-primary-500 p-xxs"
                    data-testid="icon"
                  >
                    <div
                      class="h-sm w-sm"
                    />
                  </div>
                </div>
                <span
                  data-state="closed"
                >
                  <span
                    class="text-xxs xxxl:text-xs font-medium leading-lg line-clamp-2"
                  >
                    Opportunity 2
                  </span>
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
